### ISSUE TITLE
Add a way to "rename" a headerfile in the output

### ISSUE THREAD

**Issue by jwakely** [doxygen/doxygen#9334]
This is an enhancement request.

It's common for header-only (or mostly header-only) libraries like Boost and libstdc++ to be structured in terms of a public header file that is the documented API, which includes other headers that contain the actual code:

```c++
#ifndef PUBLIC_API_H
#define PUBLIC_API_H
#include "detail/foo.h"
#include "detail/bar.h"
#endif
```

Within `detail/foo.h` there might be dozens of classes. When using the `SHOW_HEADERFILE=YES` option, those classes will show this by default in their docs:

    #include <foo.h>

That's wrong though, so every class has to have an explicit `\headerfile public_api.h` so that the docs show the right header that users should include:

    #include <public_api.h>

It would be helpful if you could put something in `detail/foo.h` once that would apply to every class within that header. Maybe change the `\file` command to take an optional second argument:

#### \file [\<name>] [\<header-file>]

Or alternatively, allow the `\headerfile` command to be used with `\file` like so:

```c++
/** \file foo.h
 *  \headerfile public_api.h
 * Implementation details. Do not include this directly, use <public_api.h> instead.
 */
```

This would then imply  `\headerfile public_api.h` for every class in the file.

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by jwakely** [comment]
Currently the format of a [`\file` command](https://www.doxygen.nl/manual/commands.html#cmdfile) is:

    \file [<name>]

and I'm suggesting that either or both of these should be allowed, with the behaviour described above:

        \file [<name>] [<header-file>]

Or:

        \file [<name>]
        \headerfile [<header-file>]

This would be consistent with [`\class`](https://www.doxygen.nl/manual/commands.html#cmdclass) which supports an optional header-file after the class-name, and also supports a `\headerfile` command in the comment block for the class. The difference would be that specifying a header-file for a file would get applied to every class in that file (unless the class has its own header-file, then that should take precedence).

**Comment by albert-github** [comment]
I just thought about this problem and in the given case there are 2 source trees (all as far as I can see):
- the original source tree with the combined / original files
- the distribution with the split files

What is there against running doxygen over the distribution, as this has also the advantage that final code is checked on complete documentation.

**Comment by jwakely** [comment]
I'm not sure what you mean. In my case, the directory hierarchy and the files are the same in the source tree and the installation tree. There is no step where original "combined" files are turned into "split" files. They're always the same files.

**Comment by albert-github** [comment]
I probably had it the wrong way around.

You have just one set of files and the user should use `public_api.h` whilst in the the `public_api.h` there are a number of files included like `details/foo.h`, `details/bar.h` etc. (which is logical from a development point of view) and doxygen sees just the fact that classes are defined in the included files though for the user it is not relevant which files are included in `public_api.h` as he just has to / should use `public_api.h` and thus see just `public_api.h`.

Would it be possible / thinkable that one included file is included in  multiple "public" files?

**Comment by jwakely** [comment]
> for the user it is not relevant which files are included in `public_api.h` as he just has to / should use `public_api.h` and thus see just `public_api.h`.

Exactly.

> Would it be possible / thinkable that one included file is included in multiple "public" files?

It's possible, but for any given class there should be a single header that is the documented way to use the class. For example, `bits/basic_string.h` is included in many parts of libstdc++ because lots of things depend on `std::string`. But the only correct way to use `std::string` is to include `<string>`. The public API that is being documented using doxygen just needs to say `#include <string>`.

I think most libraries are the same: for a given class there is one correct way to get its declaration.

**Comment by jwakely** [comment]
> Or alternatively, allow the `\headerfile` command to be used with `\file` like so:

That's related to my older #8682

**Comment by albert-github** [comment]
@doxygen 
Is this issue to be considered "fixed but not released" seen commit [989ea6d](https://github.com/doxygen/doxygen/commit/989ea6daa0702bed3f03748f05f224af540af5dd)?

**Comment by doxygen** [comment]
Yes, I see I forgot to add the usual request:
@jwakely Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by albert-github** [comment]
When I'm not mistaken it is possible that a namespace is distributed over multiple files, like

file **aa.h**
```
/// \file

/// The namespace
namespace NS
{
  /// A aa class
  class clsaa{};
}
```

file **bb.h**
```
/// \file

/// The namespace
namespace NS
{
  /// A bb class
  class clsbb{};
}
```

Though in the output I see:

<img width="403" height="441" alt="Image" src="https://github.com/user-attachments/assets/925d1cd2-3d84-445c-af8c-8e2d77a4d79c" />

so just 1 file.

Example: [example.tar.gz](https://github.com/user-attachments/files/32300333/example.tar.gz)

**Comment by doxygen** [comment]
@albert-github Indeed, my intention was to show the `#include` for namespace only when an explicit `\headerfile` was used, but it now always used the first file passed as input unless overwritten. Should be corrected now.

**Comment by albert-github** [comment]
@doxygen 
I tried a few constructs but have not been able to get any header file mentioned with the namespce for the example I provided before (so I must be missing something).
- How should this example look like?

**Comment by albert-github** [comment]
I added to both the namespace comment blocks now:
**aa.h**
```
@headerfile aa.h "aa.h"
```
and to **bb.h**
```
@headerfile bb.h "bb.h"
```

But I get:

<img width="355" height="400" alt="Image" src="https://github.com/user-attachments/assets/9d0aa430-41d8-4ab6-ac7a-33714a303a4e" />

So looks like the last one.
(the need for the `""`  part signals also that description / implementation of `\headerfile` is not consistent / not correct)
